### DISCUSSION TITLE
The DX Cluster window

### DISCUSSION THREAD

**Discussion by rsaue** [foldynl/QLog#145] [Ideas]
Hi @foldynl  

Thank you for version 0.17.  

I have some ideas for future improvement of the QLog DXcluster function.  

On my Linux desktop using Cqrlog , I have very few filtering options in the cluster section of the log program, doing most of the filtering on the dx cluster side. Using the VE7CC.Net cluster this is doable, and  that might be your design philosophy on QLog too, reducing the load from the skimmer spots on the program (?)  

![qlog - dxcluster](https://user-images.githubusercontent.com/22942440/208544928-b3a379ab-b00e-4ca6-a15e-10c26544ce4c.png)

However doing all the filtering in the DX Cluster function of Qlog could give the user the opportunity to have different filtering options in the DX cluster windows and the Bandmap window. Showing really new ones in the DX Cluster window and all traffic in the band map windows. Showing all traffic in the Band map window, gives he operator a hint of the identity of the traffic when tuning through the band (especially on CW).   

Anyway I hope you will expand the Dx Cluster functionality in QLog with the following: 

1. Expand the colours to also include worked but not confirmed band and modes,

2. Expand the filtering section with option to show only needed spots (band/mode),
 
3. When a station is reported several times on the same QRG, include a filtering option to show only the latest spot on The Dx cluster screen. Take HL5IVL i the screenshot above as an example. 

4. Use TABS to filter view of all bands or specific bands  (user selectable). Using Tabs seems to be in line with the design of other parts of QLog, e.g. the QSO logging section. Drop down menus are more cumbersome. See example from LogHX using Tabs-menu. 

 
![loghx-cluster-tabs](https://user-images.githubusercontent.com/22942440/208544897-aa5ea15d-97b0-4b5c-aa4f-7c4cf4a46d15.png)

73 de LB2EG Richard

**Comment by foldynl** [comment]
> However doing all the filtering in the DX Cluster function of Qlog could give the user the opportunity to have different filtering options in the DX cluster windows and the Bandmap window. Showing really new ones in the DX Cluster window and all traffic in the band map windows. Showing all traffic in the Band map window, gives he operator a hint of the identity of the traffic when tuning through the band (especially on CW).
> 

Unfortunately, I do not want to introduce a different filter for DX Cluster window and BandMap because it can be confusing. the current design is such that what you can see in DXC is displayed in Bandmap. Different filters in different windows could, in my opinion, confuse the operator. 

if you want to catch requested stations, use DXC Alerting. https://github.com/foldynl/QLog/wiki/Alerts

DXC Alerting is highly recommended in case of RBN.

> Anyway I hope you will expand the Dx Cluster functionality in QLog with the following:
> 
>     1. Expand the colours to also include worked but not confirmed band and modes,

this needs to be well thought out. RBN can generate huge amounts of spots (the rate can be around 40 QSO/s !!!). The performance is critical for design. I don't see, in the case of Log with 100k QSOs, how this would be done. But I'm not saying it's impossible.  

I'm leaving it open.


> 
>     2. Expand the filtering section with option to show only needed spots (band/mode),

I do not understand you exactly. DXC Filter is already present for DX Cluster window.

> 
>     3. When a station is reported several times on the same QRG, include a filtering option to show only the latest spot on The Dx cluster screen. Take HL5IVL i the screenshot above as an example.

I think that It is not good idea. with this approach you will be deprived of an overview from which parts of the world the station was heard. This can give you an overview about conditions. 

> 
>     4. Use TABS to filter view of all bands or specific bands  (user selectable). Using Tabs seems to be in line with the design of other parts of QLog, e.g. the QSO logging section. Drop down menus are more cumbersome. See example from LogHX using Tabs-menu.
> 
we will see...good idea

> 
> ![loghx-cluster-tabs](https://user-images.githubusercontent.com/22942440/208544897-aa5ea15d-97b0-4b5c-aa4f-7c4cf4a46d15.png)
> 
> 73 de LB2EG Richard

**Reply by rsaue** [reply]
Hi @foldynl,

to clarify my item 2, (Expand the filtering section with option to show only needed spots (band/mode), the QLlog filter option window does not today give the user an option to exlude from view all spots that are NOT coloured (needed).

to clarify my item 3 (show spot only once), this is proposed as a menu option, giving the user a choice. 

73 de LB2EG Richard

**Reply by foldynl** [reply]
>to clarify my item 2, (Expand the filtering section with option to show only needed spots (band/mode), the QLlog filter option window does not today give the user an option to exlude from view all spots that are NOT coloured (needed).

Oh, yes. you are right. There is missing the filter

>to clarify my item 3 (show spot only once), this is proposed as a menu option, giving the user a choice.

it makes sense to me, introduce the de-duplication option.
